// See https://github.com/apache/pulsar/blob/master/pulsar-client/src/test/proto/Test.proto
syntax = "proto3";

package pulsar;

import "pulsar_external_message.proto";

enum TestEnum {
  SHARED = 0;
  FAILOVER = 1;
}

message SubMessage {
  string foo = 1;
  double bar = 2;
  message NestedMessage {
    string url = 1;
    string title = 2;
    repeated string snippets = 3;
  }
}

message TestMessage {
  string stringField = 1;
  double doubleField = 2;
  int32 intField = 6;
  TestEnum testEnum = 4;
  SubMessage nestedField = 5;
  repeated string repeatedField = 10;
  pulsar.external.ExternalMessage externalMessage = 11;
}
